### PR TITLE
SafePatch1-0  merge with Patch 5 branch

### PR DESCRIPTION
**Pull request by Blade2021**
Incorporate new machine protection system into next update.

Additions include compatibility and functionality of monitoring sensors for time tracking analysis to preform discontinue operation if malfunction is detected.

### CODE REVIEW

File: AssemblyAutoBoard/AssemblyAutoBoard.ino [hunk 1]
```diff
@@ -195,40 +193,39 @@ void loop() {
   //Main Timer to keep track of entire machine!
   unsigned long currentTime = millis();
   //Call LCD Clear function to clear 4th line of LCD
-  lcdClear();
+  lcdControl();
```
Comment (Blade2021) [inline AssemblyAutoBoard/AssemblyAutoBoard.ino]: Went to lcdControl to provide support for RunTime function while inside the mpsInput function.

File: AssemblyAutoBoard/AssemblyAutoBoard.ino [hunk 2]
```diff
@@ -263,228 +256,260 @@ void loop() {
...
+            digitalWrite(errorLed, HIGH);
+          }
+          if (hookCheck == LOW) {
+            previousTimer3 = currentTime;
+            digitalWrite(solenoidArray[1], HIGH);
+            hookNext = 1;
+          }
         }
-        if (HookCheck == LOW) {
+        if ((currentTime - previousTimer3 < sysArray[6]) && (currentTime - previousTimer3 >= buttonWait) && (mpsEnable >= 4)) {
```
Comment (arthus3) [inline AssemblyAutoBoard/AssemblyAutoBoard.ino]: Should consider using a separate changeable variable for the sensor ignore period.

File: AssemblyAutoBoard/AssemblyAutoBoard.ino [hunk 3]
```diff
@@ -263,228 +256,260 @@ void loop() {
...
-            digitalWrite(ErrorLed, LOW);
-            FeedNext = 1;
+            digitalWrite(panelLed1, HIGH);
+            digitalWrite(errorLed, LOW);
+            feedNext = 1;
+            previousTimer1 = currentTime;
+          }
+        }
+        if (mpsEnable >= 1) {
+          if ((currentTime - previousTimer1 <= sysArray[6]) && (currentTime - previousTimer1 >= buttonWait) && (manualFeed == LOW)) {
```
Comment (arthus3) [inline AssemblyAutoBoard/AssemblyAutoBoard.ino]: Should consider using a separate changeable variable for the sensor ignore period.  
(sysArray[8])

File: AssemblyAutoBoard/AssemblyAutoBoard.ino (hunk 2 above)
Comment (Blade2021) [inline AssemblyAutoBoard/AssemblyAutoBoard.ino]: Fixed in commit #3df199e

File: AssemblyAutoBoard/AssemblyAutoBoard.ino (hunk 3 above)
Comment (Blade2021) [inline AssemblyAutoBoard/AssemblyAutoBoard.ino]: Fixed in commit: 3df199e

**Comment by Blade2021** [comment]
Issue #17 was found and fixed inside of this patch.

**Pull request merged**